### ISSUE TITLE
Add `aria-label` attributes to tabs or buttons in the UI that lack text content

### ISSUE THREAD

**Issue by Fyorl** [foundryvtt/foundryvtt#9751]

**Comment by silvative** [comment]
The issue specifically affects a huge number of buttons which constitute a single <a> element containing an item and a tooltip declared by a data-tooltip attribute. These tooltips are invisible to the screen reader which means these buttons are completely blank and unlabelled. As a solution, I have create a [branch](https://github.com/foundryvtt/foundry-vtt/tree/fix/9751) which adds aria-label attributes to some of the most immediately important of these buttons that I could find. Specifically, these are:

### HUD

- Added aria-label attribute to all canvas controls and tools
  - Added role="tablist" to the parent ol of the controls buttons
  - Added role="tab" attribute to controls buttons
  - Added role="button" attribute to tools buttons
  - Added role="tabpanel" attribute to the tools list and gave them an ID, "toolsPanel", which the control buttons point to using the aria-controls attribute.
  
- Added aria-label and role="button" attributes to the following buttons on the hotbar: Collapse, open directory, lock bar, page up, page down
   - For hotbar slots containing a macro the label is the macro's name
   
- Added aria-label and role="button" to the scene navbar toggle button
  
- Players list: 
  - Added label and role="button" to the h3 that contains the title of the players widget and also shows/hides them
  - Added tabindex attribute to the players list so that it can be navigated to
  - Added Players.List localisation string

### Sidebar

- Header buttons
  - Added role="tablist" to the parent nav
  - Labelled & added role="tab" & aria-controls attributes to the tab buttons
    - (Unsuccessfully?) added role="tabpanel" & id to the tabs themselves

- Chat tab 
  - Labelled and added button roles to "Jump to bottom", chat export, chat clear buttons
    - Added a data-tooltip field for chat export and chat clear, removing the old "title" field
  - Added new localisation string CHAT.RollVisibility for the roll mode dropdown
  - Labelled chatmessage text area
  - Labelled the chat message "delete" button
  
- Combat tracker
  - Changed existing label on the nav element
  - Labeled and added roles to create combat, previous encounter, next encounter, end combat, rollall, rollNPC, initiative reset, scope, combat settings, togglevisibility, toggledefeated, pingcombatant, rollinitiative, prev round, prev turn, end combat, next turn, next round, start combat buttons
  
- Compendium directory & Document directories & Playlist directory
  - Added roles and labels to filter, sort, and collapse all buttons
    - Added new Compendium.FilterDocuments and SIDEBAR.SortModeToggle localisation strings
  - The "Filter" button is now a <a> element like the others, instead of a naked <i>

- Playlist directory:
  - Added labels to the range inputs for playlist, ambient, and interface volume

-----

This is only a first pass and there are likely far more of these issues present in several other locations throughout various other applications and menus. For instance, I did not yet touch the token HUD. In order to change more fields in more places, we could consider modifying our data-tooltip attribute to also function as an aria-label, or automatically create a label when we detect one? This is the "lazy" approach I took in my module, though the manual approach I took above does have upsides.

### Remaining issues: 

**Big challenges**:

- As part of a future effort, we could consider changing many of these <a> elements to <button>. This would allow us to remove the role attributes without negatively affecting screen readers. There are other places where we may be able to adopt more semantic HTML, which I have noted here.
  - There is an H3 used as both header and button in the player list where a button and label combo would be more appropriate.
  - The headers for the playlist volume sliders could be labels
- Some elements felt awkward to access or use with the screen reader.
  - The 'Clear chat' button did not seem easy to find, it's inside a shared div with the "export chat log" button and isn't announced
  - When a new scene control group is selected, it might be good to jump to the tools, or to move the tools list up in the HTML so that it's immediately below the control group's button, in order to make it easier to find. Currently there is no feedback that the buttons later on have changed (though labelling them certainly helps).
    - Is there a way that we can tell users which group is currently active? At the very least appending "active" to the label when the control is active?
    - Could we just have all of the tools exist all the time, and visually hide those from inactive tabs, so that screenreader users can just navigate the entire list using the control groups as landmarks/headers?
  - Similarly, the dropdown for the compendium filters is inserted at an unintuitive location in the HTML which means it's not easy to tell that it's been opened at all.
  - The descriptions of each of the volume sliders on the playlists tab cannot currrently be accessed via screen reader as this information exists only within the data-tooltip attribute
  
Since these changes would meaningfully modify our CSS/HTML they are probably out of the question until V12 at least.
  
**Easy challenges:**
 
There were also a few small changes that perhaps could have fit into the scope of this commit but that I was not confident in making.

- I didn't make the label for the macro bar lock/unlock button change based on whether or not it's currently locked
- I think the sidebar's 'sort mode toggle' button's tooltip was broken (at least, it was only creating a blank string). I assume it was supposed to change to reflect the current sort mode but since it wasn't working I have temporarily set both it and the label to a new value. If it is fixed or if it was never broken then we can revert that.
- Empty macro slots are not labelled with anything other than their number
- There was some odd behaviour with the "toggle search mode" button beside the sidebar search tabs that I did not fully understand, so I tried to leave it untouched under the assumption it was intentional.
  - On "document" tabs when the icon is "disabled" and hidden, the screen reader can still see it. I've added a label to it, but I didn't otherwise touch it. If this is intentional then labelling it has probably just made it worse, so we should see if we can do more to hide the button than just hiding the icon.
  - On the compendium tab the icon appears but is not a link, it's just an icon by itself. I think it may be being used as a purely visual indication that the text field next to it is a search, but it's a button on the other tabs, so this is a little confusing.
